### PR TITLE
add linear warmup scheduler

### PR DESCRIPTION
**Pull request by wetdog**
Add linear scheduler as mentioned in the model configuration section of the paper "We utilized a linear decay learning rate
schedule with a peak learning rate of 7.5 × 10−5 and incorporated a warm-up phase for the initial 20,000 updates"

### CODE REVIEW

File: e2_tts_pytorch/trainer.py [hunk 1]
```diff
@@ -154,13 +157,22 @@ def load_checkpoint(self):
...
     def train(self, train_dataset, epochs, batch_size, grad_accumulation_steps=1, num_workers=12, save_step=1000):
         # (todo) gradient accumulation needs to be accounted for
 
         train_dataloader = DataLoader(train_dataset, batch_size=batch_size, collate_fn=collate_fn, shuffle=True, num_workers=num_workers, pin_memory=True)
-        train_dataloader = self.accelerator.prepare(train_dataloader)
+        total_steps = len(train_dataloader) * epochs
+        decay_steps = total_steps - self.num_warmup_steps
+        warmup_scheduler = LinearLR(self.optimizer, start_factor=1e-8, end_factor=1.0, total_iters=self.num_warmup_steps)
+        decay_scheduler = LinearLR(self.optimizer, start_factor=1.0, end_factor=1e-8, total_iters=decay_steps)
+        self.scheduler = SequentialLR(self.optimizer,
```
Comment (lucidrains) [inline e2_tts_pytorch/trainer.py:172]: TIL! this is cool
